lightningd/payalgo.c

```diff
@@ -683,6 +687,8 @@ static void json_pay(struct command *cmd,
 	/* Start with no failures */
 	list_head_init(&pay->pay_failures);
 	pay->in_sendpay = false;
+	pay->description = tal_dup_arr(pay, char, b11->description,
+				       tal_len(b11->description), 0);
```

Comment:
`pay->description = b11->description;`

b11 is a tal child of pay, so there are no lifetime issues with this,and it handled NULL correctly.